src/docs-developer-guide.rst

```diff
@@ -122,3 +122,188 @@ For example:
   }
 
   /* New section ends */
+
+.. _style-tests:
+
+Style Guide checks
+--------------------
+
+Proselint_  is used for style testing the docs.
+Apart from the inbuilt tests in proselint,
+custom checks are added for style guide testing.
+We follow the approach of `literate programming <https://en.wikipedia.org/wiki/Literate_programming>`_.
+The checks reside in the file :doc:`docs-style-guide.rst <docs-style-guide>`.
+After each style rule,
+you can define a python code-block
+containing the code for style testing.
+When style testing is performed,
+these python code-blocks are parsed to
+generate a testing script.
+
+.. _proselint-checks:
+
+Proselint dependent checks
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Various inbuilt functions from Proselint are used to define the tests:
+
+.. py:function:: memoize
+
+  To cache the check for faster execution. Use :py:data:`@memoize` above function definition to cache the result.
```

Comment:
This is a decorator, not a function.
(I don't think there is a `py:decorator` role, so `py:function` is fine. But the prose should be corrected.)